Containers/talk/start.sh

```diff
@@ -16,43 +16,46 @@ elif [ -z "$INTERNAL_SECRET" ]; then
 fi
 
 set -x
-IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short)"
+if [ -n "$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_TALK="$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
+
+if [ -n "$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_NC="$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_NC="$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
 set +x
 
 # Turn
-cat << TURN_CONF > "/etc/turnserver.conf"
-listening-port=$TALK_PORT
-fingerprint
-use-auth-secret
-static-auth-secret=$TURN_SECRET
-realm=$NC_DOMAIN
-total-quota=0
-bps-capacity=0
-stale-nonce
-no-multicast-peers
-simple-log
-pidfile=/var/tmp/turnserver.pid
-no-tls
-no-dtls
-userdb=/var/lib/turn/turndb
-# Based on https://nextcloud-talk.readthedocs.io/en/latest/TURN/#turn-server-and-internal-networks
-allowed-peer-ip=$IPv4_ADDRESS_TALK
-denied-peer-ip=0.0.0.0-0.255.255.255
-denied-peer-ip=10.0.0.0-10.255.255.255
-denied-peer-ip=100.64.0.0-100.127.255.255
-denied-peer-ip=127.0.0.0-127.255.255.255
-denied-peer-ip=169.254.0.0-169.254.255.255
-denied-peer-ip=172.16.0.0-172.31.255.255
-denied-peer-ip=192.0.0.0-192.0.0.255
-denied-peer-ip=192.0.2.0-192.0.2.255
-denied-peer-ip=192.88.99.0-192.88.99.255
-denied-peer-ip=192.168.0.0-192.168.255.255
-denied-peer-ip=198.18.0.0-198.19.255.255
-denied-peer-ip=198.51.100.0-198.51.100.255
-denied-peer-ip=203.0.113.0-203.0.113.255
-denied-peer-ip=240.0.0.0-255.255.255.255
+cat << TURN_CONF > "/etc/eturnal.yml"
+eturnal:
+  listen:
+    - ip: "::"
+      port: $TALK_PORT
+      transport: udp
+    - ip: "::"
+      port: $TALK_PORT
+      transport: tcp
+  log_dir: stdout
+  log_level: notice
+  secret: "$TURN_SECRET"
+  relay_ipv4_addr: "$IPv4_ADDRESS_NC"
+  relay_ipv6_addr: "$IPv6_ADDRESS_NC"
```

Comment:
why do we need this and shold this not be IPv4_ADDRESS_TALK and IPv6_ADDRESS_TALK ?

Reply:
This option specifies the IPv4 address used for relaying data from/to TURN peers. Note that this must be the server's public IPv4 address
https://eturnal.net/documentation/#relay_ipv4_addr

Correspondingly to ipv6, although this is optionally. Alternatively, this can be done via ENV variables: 
https://eturnal.net/documentation/#ETURNAL_RELAY_IPV4_ADDR

Reply:
I wonder does coturn get the relay address on its own? Because with coturn we did not have to configure this manually...

Reply:
Apparently it does: 
![image](https://github.com/nextcloud/all-in-one/assets/42591237/9c7c6637-2c45-4844-9375-6dc9c74e3e4e)

Reply:
That is an internal IP address. I don't know if this is the actual IP address needed in your environment.

In our official image we have a STUN lookup to discover the external/public IP address for peers to connect.
https://github.com/processone/eturnal/tree/master/docker-k8s#general-hints

Reply:
eturnal also tries to autodetect the IP addresses: https://eturnal.net/documentation/#relay_ipv4_addr , the STUN query is an additonal part to make the auto-detection more robust for container environments.

Reply:
So if it automatically discovers the relay ip-address why do we need to add it manually?

Reply:
I [implied](https://github.com/processone/eturnal/issues/42#issuecomment-1605589548) to keep it. The STUN lookup is prepended to [eturnalctl](https://github.com/sando38/eturnal/blob/ca13a4fb4ead2768302f6c37e97144dcbafa37ec/docker-k8s/image/Dockerfile#L117-L126). It renders useless, however, if something is specified in the configuration file, as that is eturnal's truth. 

I guess adding it manually into the configuration is to ensure to have the correct address to be advertise to TURN users/ peers - as I have seen you try to cover potentially more rare use case.

So in that case it makes sense to have it specified there.

I would keep things more simple and leave it to the auto-detection. If people have more special cases - and I assume, they are aware of increased complexity if the have VPNs, etc. configured - , they should mount a [custom configuration file](https://github.com/sando38/eturnal/blob/master/docker-k8s/README.md#configuration) into the container, which covers there special needs (potentially also the whitelist). The mounting point would be `/etc/eturnal.yml`.

Your start script should target `/usr/lib/eturnal/etc/eturnal` (alpine package) or `/opt/eturnal/etc/eturnal.yml` (docker package). And if people decide to choose a custom, there is no conflict, if they mount to `/etc/eturnal.yml` as that path has precedence.

So, to understand your "80%" users:
* they usually have one Docker host in which they run there applications?
* they usually have a static IP address?

Reply:
IPv4_ADDRESS_NC is the ip adress of the nextcloud domain
IPv4_ADDRESS_TALK is the internal ip of the dockercontainer

Reply:
But just for me again, does the automatic ip check from eturnal cover dynamic ips? (so does it check the ip once or in an intervall?)

Reply:
if not then we should use a cron script to update the ip adress in the config and reload eturnal, right?

Reply:
> if not then we should use a cron script to update the ip adress in the config and reload eturnal, right?

Yes, see here: https://github.com/nextcloud/all-in-one/pull/2855#discussion_r1241064433

Reply:
do you think it would work now?

Reply:
The cronjob script looks okay.

Reply:
Provided of course, that `$NC_DOMAIN` resolves to the IP address clients can connect to. Otherwise you may choose to define a variable `TURN_DOMAIN` which defaults to `NC_DOMAIN`  and can be used for more complex cases.

Reply:
That may even solve the challenge to always pick the correct IP address in the non-dynamic case well.

Reply:
https://github.com/nextcloud/all-in-one/pull/2855#discussion_r1241233958